lib/checkbufferoverrun.cpp

```diff
@@ -1833,6 +1833,10 @@ void CheckBufferOverrun::checkInsecureCmdLineArgs()
                 if (tok->varId() == argvVarid)
                     break;
 
+                // Update varid in case the input is copied by strdup()
+                if (Token::Match(tok, "strdup ( %varid%", argvVarid) && tok->tokAt(-2))
```

Comment:
Change the Token::Match to something like:

    Token::Match(tok->tokAt(-2), "%var% = strdup ( %varid%")